core/block/object/objectgraph/graph.go

```diff
@@ -128,7 +128,12 @@ func (gr *Builder) buildGraph(
 
 		outgoingRelationLink := make(map[string]struct{}, 10)
 		edges = gr.appendRelations(rec, relations, edges, existedNodes, sourceId, outgoingRelationLink)
-		edges = gr.appendLinks(req.SpaceId, rec, outgoingRelationLink, existedNodes, edges, sourceId)
+		var nodesToAdd []*types.Struct
+		nodesToAdd, edges = gr.appendLinks(req.SpaceId, rec, outgoingRelationLink, existedNodes, edges, sourceId)
+
+		if len(nodesToAdd) != 0 {
```

Comment:
no need to do an empty check. `append` will be noop in case `nodesToAdd` is empty

Reply:
Thank you! 
Also we should map relations in details as requested in keys field of request